src/main/ruby/truffleruby/core/struct.rb

```diff
@@ -147,7 +147,7 @@ def to_s
 
       name = self.class.name
 
-      if Primitive.nil?(name) || name.empty?
+      if Primitive.nil?(name) || name.empty? || name[0] == '#'
```

Comment:
cc @andrykonchin 
I think you wanted a similar check for Marshal, maybe we can use the same way/Primitive for both cases?
For context, name[0] == '#' seems a rather inefficient way to check if a module is anonymous (i.e., not fully named).

Reply:
To be clear, I don't think we need to change that in this PR, we just need to remember to use the new way in @andrykonchin's Marshal PR which will likely be merged after this one.

Reply:
Yeah, it completely makes sense 👍 . Will take care of this.